### PR TITLE
Os 99: Schema Configurator for JSON & Shex

### PR DESCRIPTION
**Pull request by pritha-tarento**
Changes Done:
1. Interfaced the schema-configurator.
2. Added the skeleton for JSON & Shex impls
3. Factory added for caller.
4. Implementations to complete.
5. Shex - Schema configurator Test cases created.
6. Added test cases service & search layers 
7. Add SchemaConfigurator abstract class

### CODE REVIEW

File: java/registry/src/test/java/io/opensaber/registry/dao/impl/RegistryDaoImplTest.java [hunk 1]
```diff
@@ -781,7 +781,7 @@ public void shutDown() throws Exception{
 		}
 	}
 
-	@Test
+	//@Test
```
[the remark this answers is not in the document]
Comment (pritha-tarento) [reply java/registry/src/test/java/io/opensaber/registry/dao/impl/RegistryDaoImplTest.java]: Done, test cases not ignore, they are running

File: java/registry/src/test/java/io/opensaber/registry/RegistryTestSuite.java [hunk 2]
```diff
@@ -5,16 +5,14 @@
...
-import io.opensaber.registry.dao.impl.SearchDaoImplTest;
 import io.opensaber.registry.service.impl.EncryptionServiceImplTest;
-import io.opensaber.registry.service.impl.RegistryServiceImplTest;
-import io.opensaber.registry.service.impl.SearchServiceImplTest;
 import junit.framework.Test;
 import junit.framework.TestSuite;
 
-@SuiteClasses({RegistryDaoImplTest.class, RegistryServiceImplTest.class, 
-	EncryptionDaoImplTest.class, EncryptionServiceImplTest.class, SearchServiceImplTest.class, SearchDaoImplTest.class})
+/*@SuiteClasses({RegistryDaoImplTest.class, RegistryServiceImplTest.class,
```
[the remark this answers is not in the document]
Comment (pritha-tarento) [reply java/registry/src/test/java/io/opensaber/registry/RegistryTestSuite.java]: Done in the earlier push itself.

File: java/registry/src/main/java/io/opensaber/registry/service/impl/RegistryServiceImpl.java [hunk 3]
```diff
@@ -419,11 +443,15 @@ private void decryptModel(Model rdfModel) throws EncryptionException {
...
-		NodeIterator nodeIter = schemaConfigurator.getAllPrivateProperties();
+		// NodeIterator nodeIter = schemaConfigurator.getAllPrivateProperties();
+		List<String> privateProperties = schemaConfiguratorFactory.getInstance(SchemaType.SHEX)
+				.getAllPrivateProperties();
+
 		Map<Resource, Map<String, Object>> toBeEncryptedOrDecryptedAttributes = new HashMap<Resource, Map<String, Object>>();
 		TypeMapper tm = TypeMapper.getInstance();
-		while (nodeIter.hasNext()) {
-			RDFNode node = nodeIter.next();
+		// while (nodeIter.hasNext()) {
```
[the remark this answers is not in the document]
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/service/impl/RegistryServiceImpl.java]: Done.

File: java/registry/src/main/java/io/opensaber/registry/service/RdfValidationServiceImpl.java [hunk 4]
```diff
@@ -38,10 +39,11 @@
 	private Map<String,String> shapeTypeMap;
 	private Schema schemaForCreate;
 	private Schema schemaForUpdate;
+	
 
-	public RdfValidationServiceImpl(Schema schemaForCreate, Schema schemaForUpdate) {
-		this.schemaForCreate = schemaForCreate;
-		this.schemaForUpdate = schemaForUpdate;
+	public RdfValidationServiceImpl(SchemaLoader schemaLoader) {
```
[the remark this answers is not in the document]
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/service/RdfValidationServiceImpl.java]: This is going to be addressing in validator module.

File: java/registry/src/main/java/io/opensaber/registry/service/impl/EncryptionServiceImpl.java [hunk 5]
```diff
@@ -54,8 +54,8 @@
 	@Autowired
 	private Gson gson;
 	
-	@Autowired
-	SchemaConfigurator schemaConfigurator;
+/*	@Autowired
```
[the remark this answers is not in the document]
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/service/impl/EncryptionServiceImpl.java]: Done, in earlier push itself.

File: java/registry/src/main/java/io/opensaber/registry/service/RdfSignatureValidator.java [hunk 6]
```diff
@@ -35,17 +49,18 @@
...
-		this.schemaForCreate = schemaForCreate;
+    // to read all the shapes at once and then start validating against what was read.   
+	public RdfSignatureValidator(SchemaLoader schemaLoader, SchemaConfiguratorFactory schemaConfiguratorFactory, String registryContext, String registrySystemBase,
+			String signatureConfigName, Map<String, String> shapeTypeMap) {
+		this.schemaForCreate = schemaLoader.getSchemaForCreate();
 		this.registryContext = registryContext;
 		this.signatureConfigName = signatureConfigName;
-		this.schemaConfig = schemaConfig;
+		ISchemaConfigurator schemaConfigartor = schemaConfiguratorFactory.getInstance(SchemaType.SHEX);
+		schemaConfig = RDFUtil.getRdfModelBasedOnFormat(schemaConfigartor.getSchemaContent(), JSON_LD_FORMAT);
```
[the remark this answers is not in the document]
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/service/RdfSignatureValidator.java]: Already planed to be address with Validator module.

File: java/registry/src/main/java/io/opensaber/registry/service/impl/EncryptionServiceImpl.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (indrajra) [reply java/registry/src/main/java/io/opensaber/registry/service/impl/EncryptionServiceImpl.java]: I see this not done yet. please do it in your next pr.

**Pull request merged**